File: src/main/java/xyz/chunkstories/api/plugin/PluginInformation.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.plugin;

import java.io.File;
import java.util.Collection;

import xyz.chunkstories.api.GameContext;
import xyz.chunkstories.api.exceptions.plugins.PluginCreationException;
import xyz.chunkstories.api.plugin.commands.PluginCommand;

public interface PluginInformation {
	public String getName();

	public String getPluginVersion();

	public String getAuthor();

	public PluginType getPluginType();

	public ChunkStoriesPlugin createInstance(GameContext pluginExecutionContext) throws PluginCreationException;

	public Collection<PluginCommand> getCommands();

	public File getDirectory();

	public enum PluginType
	{
		UNIVERSAL, CLIENT_ONLY, MASTER;
	}
}

File: src/main/java/xyz/chunkstories/api/plugin/PluginInformation.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.plugin

data class PluginInformation(
        val name: String,
        val version: String = "1.0",
        val authors: String,
        val pluginType: PluginType = PluginType.UNIVERSAL,
        val entryPoint: String
) {

    enum class PluginType {
        UNIVERSAL, CLIENT_ONLY, MASTER
    }
}